preliz/ppls/pymc_io.py

```diff
@@ -201,11 +199,51 @@ def reshape_params(model, var_info, p_model, params):
     return value
 
 
-def non_constant_parents(var_):
+def non_constant_parents(var_, free_rvs):
     """Find the parents of a variable that are not constant."""
     parents = []
     for variable in var_.get_parents()[0].inputs[2:]:
         if not isinstance(variable, TensorConstant):
-            parents.append(variable.owner.inputs[0])
-
+            for free_rv in free_rvs:
+                if free_rv in list(ancestors([variable])) and free_rv not in parents:
+                    parents.append(free_rv)
     return parents
+
+
+def posterior_to_prior(model, posterior, alternative=None):
+    """
+    Get the model information for fitting the samples from prior into user provided model's prior.
+    We need to "backfit" because we can not use arbitrary samples as priors.
+
+    Parameters
+    ----------
+    model : A PyMC model
+    A probabilistic model
+
+    posterior : Posterior samples
+    InferenceData from with the posterior group
+
+    alternative : "auto", list, dict, defaults to None
+    Users can add the model variables to consider alternative distributions while fitting samples
+
+    """
+
+    model_info = get_model_information(model)[2]
+    parsed_info = [(dist, var) for var, dist in model_info.items()]
+    new_priors = []
+
+    for dist, var in parsed_info:
+        dists = [model_info[var]]
+
+        if alternative == "auto":
+            dists += [pz.Normal(), pz.HalfNormal(), pz.Gamma()]
+        elif isinstance(alternative, list):
+            dists += alternative
+        elif isinstance(alternative, dict):
+            dists += alternative.get(var, [])
+        # Take the dist with the least penalization term
+        idx = pz.mle(dists, posterior[var].values)[0]
```

Comment:
Pass plot=False. Also while this work for a single distribution, in that case it is better to just use the private method fit_mle, it should be slightly faster.

Reply:
Are you talking about the case when alternative="None", that we should use fit_dist and for other cases use pz.mle(when more than 1 distributions are there)
Since, pz.mle gives us an array with the least loss and we pick the least one?

Reply:
Yes, when we only have a single distribution "alternative=None" there is not reason to use use pz.mle other that making the code slightly simpler

Reply:
Thanks have made the changes when `len(dists) = 1`